### DISCUSSION TITLE
Can I change the authors with -M via command line ?

### DISCUSSION THREAD

**Discussion by KaiWaldrant** [quarto-dev/quarto-cli#6260] [Feature Requests]
### Description

I was wondering if you can change the author metadata like the title and subtitle metadata via the command line option `-M`.

We are creating dynamic html reports with quarto but the authors should change depending on who has setup the analysis. So I was trying to set the author field dynamically with `-M author:"Kai Waldrant"` like I did with the title and subtitle `-M title:"Test title" -M subtitle:"Test subtitle"` but only the titles changes not the author. Is there a way to do this ?

I have been searching the discussion part but haven't found an answer.

**Comment by mcanouil** [comment]
I am not sure you can as the `author` field is not a scalar, see <https://quarto.org/docs/journals/authors.html#author-schema>.

Pinging @dragonstyle in case I am mistaken.

**Comment by rgaiacs** [comment]
`--variable` and `--metadata` do **not** work based on the metadata reported by Quarto's log.

`--metadata-file` works based on the metadata reported by Quarto's log but this does **NOT** show in the final output.

# Minimal Working Example

`mwe.qmd`:

```
---
title: MWE
author: Alice
---

Foo
```

`metadata.yaml`:

```
author: Eve
```

`quarto render mwe.qmd --to html --metadata-file=metadata.yaml` prints

```
pandoc 
  to: html
  output-file: mwe.html
  standalone: true
  section-divs: true
  html-math-method: mathjax
  wrap: none
  default-image-extension: png
  
metadata
  document-css: false
  link-citations: true
  date-format: long
  lang: en
  author: Eve
  title: MWE
  
Output created: mwe.html
```

`grep -i alice mwe.html` prints

```
<meta name="author" content="Alice">
             <p>Alice </p>
```

**Reply by KaiWaldrant** [reply]
Yeah so that's what I am experiencing. I find it weird that you can change the author field in that way but the output still has the previous value. What is the point of this then ?

**Reply by cderv** [reply]
This could be related to this issue: 
* https://github.com/quarto-dev/quarto-cli/issues/6123

> --metadata values will be processed through isQuartoMetadata() function, and add to metadata to be used by Quarto process. Otherwise they will be considered Pandoc only metadata.

`author` is among the value that are not correctly identified as `isQuartoMetadata()`. It is why using `-M` does not work here. 
Though you wouldn't be able to pass only a string for author and not another object using `-M`

>`--metadata-file` is only considered as Pandoc only metadata. No quarto metadata (like citation or date-format) can be passed to be processed by Quarto

Authors field needs to be treated by Quarto in a specific way, and I don't think this works with `--metadata-files` currently. 

But also `--metadata-files` in Pandoc does not override YAML header field value  (https://pandoc.org/MANUAL.html#option--metadata-file)
> Metadata values specified inside the document, or by using [-M](https://pandoc.org/MANUAL.html#option--metadata), overwrite values specified with this option.

So it can't be use even with Pandoc to change the value. This means we either

* have a report problem in the Quarto log as `author: Eve` should not be shown 
* or we do change the Pandoc behavior and override and this is really #6123 problem. 

cc @dragonstyle in case you have more context. We should definitely make all that clearer.

**Comment by KaiWaldrant** [comment]
I changed the discussion to a feature request.

**Comment by cderv** [comment]
> We are creating dynamic html reports with quarto but the authors should change depending on who has setup the analysis. So I was trying to set the author field dynamically with -M author:"Kai Waldrant" like I did with the title and subtitle -M title:"Test title" -M subtitle:"Test subtitle" but only the titles changes not the author. Is there a way to do this ?

Have you considered making your file use an external file metadata ? 
https://quarto.org/docs/projects/quarto-projects.html#shared-metadata

This would be a hardcoded file name in your `.qmd` but the content would change based on who is the author ? 

For a project with a dir like that
````
Quarto project
--| _quarto.yml
--| article_dir
   --| doc.qmd
   --| _medadata.author1.yml
   --| _medadata.author2.yml
```` 

You could do a [pre render script](https://quarto.org/docs/projects/scripts.html#pre-and-post-render) or any other one (like makefile) and do 

```
cp article_dir/_metadata.author1.yml article_dir/_metadata.yml
quarto render doc.qmd
```

This examples leverage the directory shared meta (https://quarto.org/docs/projects/quarto-projects.html#directory-metadata) but you could also leverage the `metadata-file` YAML option: https://quarto.org/docs/projects/quarto-projects.html#metadata-includes
with the same concept. 

````markdown
---
title: MWE
metadata-file: authors.yml
format: html
---

Foo
```` 

You can then adapt `authors.yml` content depending on who renders. 

It seems this works even if `--metadata-files` don't because it is not processed the same... 🤔 Related to https://github.com/quarto-dev/quarto-cli/discussions/6260#discussioncomment-6476525 and #6123 

You can't do 
````
quarto render doc.qmd -M metadata-file:author-1.yml
````
because of https://github.com/quarto-dev/quarto-cli/issues/6123 

see my previous comment https://github.com/quarto-dev/quarto-cli/discussions/6260#discussioncomment-6476525

Definitely something we need to improve here

**Reply by KaiWaldrant** [reply]
i'll check this out

**Reply by KaiWaldrant** [reply]
It does work. Thank you for the workaround.

Also checked if it would add an author to the output if the author field was already defined. It falls back to the method of the --metadata-file saga. Looks like this is all a bit underdeveloped (which I understand, no accusations here :) ) We're all here to test and report 

```yaml
---
author:
  - name: author 1
  - name: author 2
metadata-file: authors.yaml
---
```
The authors defined in `authors.yaml` is not in the output html but is added in the pandoc metadata

**Reply by cderv** [reply]
> if it would add an author to the output if the author field was already defined. It falls back to the method of the --metadata-file saga.

I believe this is the Pandoc default behavior here. As I said in other post above,  `--metadata-files` in Pandoc does not override YAML header field value ([pandoc.org/MANUAL.html#option--metadata-file](https://pandoc.org/MANUAL.html#option--metadata-file))

When using `metadata-file`, it requires to have no YAML field in the document that would override it. 

I am not sure we would want to change this Pandoc behavior 🤔 Or if we do we would need to really warn and document it.

**Reply by KaiWaldrant** [reply]
Either way the documentation needs some clarifying
